pyanaconda/ui/gui/spokes/advstorage/fcoe.py

```diff
@@ -85,53 +82,48 @@ def use_dcb(self):
         return self._autoCheckbox.get_active()
 
-    def _add(self):
-        try:
-            self._addError = self.fcoe.add_san(self.nic, self.use_dcb, self.use_auto_vlan)
-        except (IOError, OSError) as e:
-            self._addError = str(e)
-
-        self._after_add()
-
-    @async_action_wait
-    def _after_add(self):
-        # When fcoe discovery is done, update the UI.  We don't need to worry
-        # about the user escaping from the dialog because all the buttons are
-        # marked insensitive.
-        self._addSpinner.stop()
-        self._addSpinner.hide()
-
-        for widget in [self._addButton, self._cancelButton, self._nicCombo,
-                       self._dcbCheckbox, self._autoCheckbox]:
-            widget.set_sensitive(True)
-
-        if self._addError:
-            # Failure.  Display some error message and leave the user on the
-            # dialog to try again.
-            self.builder.get_object("errorLabel").set_text(self._addError)
-            self._errorBox.set_visible(True)
-            self._errorBox.set_no_show_all(False)
-            self._errorBox.show()
-
-            self._addError = None
-            self._addButton.set_sensitive(True)
-            self._cancelButton.set_sensitive(True)
-        else:
-            # Success.  There's nothing else the user can do on this dialog.
-            self.fcoe.added_nics.append(self.nic)
-            self._update_devicetree = True
-            self.window.response(1)
+    def _set_configure_sensitive(self, sensitivity):
+        """Set widgets to the given sensitivity."""
+        self._addButton.set_sensitive(sensitivity)
+        self._cancelButton.set_sensitive(sensitivity)
+        self._nicCombo.set_sensitive(sensitivity)
+        self._dcbCheckbox.set_sensitive(sensitivity)
+        self._autoCheckbox.set_sensitive(sensitivity)
 
     def on_add_clicked(self, *args):
-        # Set some widgets to visible/not while we work.
+        """Start the discovery task."""
+        # First update widgets.
+        self._set_configure_sensitive(False)
         self._errorBox.hide()
-        self._addSpinner.set_visible(True)
-        self._addSpinner.show()
 
-        for widget in [self._addButton, self._cancelButton, self._nicCombo,
-                       self._dcbCheckbox, self._autoCheckbox]:
-            widget.set_sensitive(False)
+        # Get the discovery task.
+        task_path = self._fcoe_proxy.DiscoverWithTask(self.nic, self.use_dcb, self.use_auto_vlan)
+        task_proxy = STORAGE.get_proxy(task_path)
+
+        # Start the discovery.
+        async_run_task(task_proxy, self.process_result)
+
+        self._spinner.start()
+        self._spinner.show()
+
+    def process_result(self, task_proxy):
+        """Process the result of the task.
 
-        self._addSpinner.start()
+        :param task_proxy: a task proxy
+        """
+        # Stop the spinner.
+        self._spinner.stop()
+        self._spinner.hide()
```

Comment:
I would guess that you want to stop the spinner on the end of this method.

Reply:
This is same as in the dialog window for zfcp and dasd.

Reply:
Yeah it should be fine on both sides because, if I'm not mistaken, this is happening when event loop is idle. So it is more about preference, personally I would move that to the end of the method because it will happen after the method but I'm fine with both.